indexer/block.go

```diff
@@ -1,14 +1,134 @@
 package indexer
 
-import "github.com/photon-storage/go-photon/chain/gateway"
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"gorm.io/gorm"
+
+	"github.com/photon-storage/go-photon/chain/gateway"
+
+	"github.com/photon-storage/photon-explorer/database/orm"
+)
 
 func (e *EventProcessor) processBlock(block *gateway.BlockResp) error {
-	// TODO(doris)
+	if err := e.db.Transaction(func(tx *gorm.DB) error {
+		b := &orm.Block{
+			Slot:              block.Slot,
+			Hash:              block.BlockHash,
+			ParentHash:        block.ParentHash,
+			StateHash:         block.StateHash,
+			ProposalIndex:     block.ProposerIndex,
+			ProposalSignature: block.ProposerSignature,
+			RandaoReveal:      block.RandaoReveal,
+			Graffiti:          block.Graffiti,
+			Timestamp:         block.Timestamp,
+		}
+
+		if err := tx.Model(&orm.Block{}).Create(b).Error; err != nil {
+			return err
+		}
+
+		if err := processAttestations(tx, b.ID, block.Attestations); err != nil {
+			return err
+		}
+
+		return e.processTransactions(tx, b.ID, block.Txs)
+	}); err != nil {
+		return err
+	}
+
 	e.currentSlot = block.Slot
 	e.currentHash = block.BlockHash
 	return nil
 }
 
+func processAttestations(db *gorm.DB, blockID uint64, attestations []*gateway.Attestation) error {
+	as := make([]*orm.Attestation, len(attestations))
+	for i, a := range attestations {
+		bits := strings.Replace(
```

Comment:
Strings.Replace is a bit hacky.
https://stackoverflow.com/questions/37532255/one-liner-to-transform-int-into-string